core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -0,0 +1,553 @@
+                    .map_err(|_| VerificationError::ServiceManagerError)?,
+            ),
+            data: Some(zksync_basic_types::web3::Bytes(data)),
+            ..Default::default()
+        };
+
+        let res = self
+            .signing_client
+            .as_ref()
+            .call_contract_function(call_request, None)
+            .await
+            .map_err(|_| VerificationError::ServiceManagerError)?;
+
+        self.decode_bytes(res.0.to_vec())
+            .map_err(|_| VerificationError::ServiceManagerError)
+    }
+
+    /// Verifies that the certificate's blob quorum params are correct
+    pub async fn verify_security_params(&self, cert: BlobInfo) -> Result<(), VerificationError> {
+        let blob_header = cert.blob_header;
+        let batch_header = cert.blob_verification_proof.batch_medatada.batch_header;
+
+        let mut confirmed_quorums: HashMap<u32, bool> = HashMap::new();
+        for i in 0..blob_header.blob_quorum_params.len() {
+            if batch_header.quorum_numbers[i] as u32
+                != blob_header.blob_quorum_params[i].quorum_number
+            {
+                return Err(VerificationError::WrongQuorumParams);
+            }
+            if blob_header.blob_quorum_params[i].adversary_threshold_percentage
+                > blob_header.blob_quorum_params[i].confirmation_threshold_percentage
+            {
+                return Err(VerificationError::WrongQuorumParams);
+            }
+            let quorum_adversary_threshold = self
+                .get_quorum_adversary_threshold(blob_header.blob_quorum_params[i].quorum_number)
+                .await?;
+
+            if quorum_adversary_threshold > 0
+                && blob_header.blob_quorum_params[i].adversary_threshold_percentage
+                    < quorum_adversary_threshold as u32
+            {
+                return Err(VerificationError::WrongQuorumParams);
+            }
+
+            if (batch_header.quorum_signed_percentages[i] as u32)
+                < blob_header.blob_quorum_params[i].confirmation_threshold_percentage
+            {
+                return Err(VerificationError::WrongQuorumParams);
+            }
+
+            confirmed_quorums.insert(blob_header.blob_quorum_params[i].quorum_number, true);
+        }
+
+        let required_quorums = self.call_quorum_numbers_required().await?;
+
+        for quorum in required_quorums {
+            if !confirmed_quorums.contains_key(&(quorum as u32)) {
+                return Err(VerificationError::QuorumNotConfirmed);
+            }
+        }
+        Ok(())
+    }
+
+    /// Verifies that the certificate is valid
+    pub async fn verify_inclusion_data_against_settlement_layer(
+        &self,
+        cert: BlobInfo,
+    ) -> Result<(), VerificationError> {
+        self.verify_batch(cert.clone()).await?;
+        self.verify_merkle_proof(cert.clone())?;
+        self.verify_security_params(cert.clone()).await?;
```

Comment:
I'm reasonably sure `cert` can be passed by ref at least for the last two methods (`verify_batch` is tricky because of `ethabi::Token`s created in `hash_batch_metadata`). In general, please pass values by ref where possible; there are too many examples in this module and adjacent modules to highlight.

Reply:
fixed here https://github.com/lambdaclass/zksync-era/pull/375